src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -932,7 +932,8 @@ private void EmitArguments(ImmutableArray<BoundExpression> arguments, ImmutableA
             // We might have an extra argument for the __arglist() of a varargs method.
             Debug.Assert(arguments.Length == parameters.Length || arguments.Length == parameters.Length + 1, "argument count must match parameter count");
             Debug.Assert(parameters.All(p => p.RefKind == RefKind.None) || !argRefKindsOpt.IsDefault, "there are nontrivial parameters, so we must have argRefKinds");
-            Debug.Assert(argRefKindsOpt.IsDefault || argRefKindsOpt.Length == arguments.Length, "if we have argRefKinds, we should have one for each argument");
+            // We might have a missing ref kind for the __arglist() of a varargs method.
+            Debug.Assert(argRefKindsOpt.IsDefault || argRefKindsOpt.Length == arguments.Length || argRefKindsOpt.Length == arguments.Length - 1, "if we have argRefKinds, we should have one for each argument");
```

Comment:
>argRefKindsOpt.Length == arguments.Length - 1 [](http://example.com/codeflow?start=98&length=45)

Let's strengthen the assert - ```(argRefKindsOpt.Length == arguments.Length - 1 && arguments.Last() is BoundArgListOperator)```. And let's strengthen the first assert in a similar way.

Reply:
FWIW, this is already asserted below in `GetArgumentRefKind`:

https://github.com/dotnet/roslyn/blob/17dcec138afd78a265be020ef8ca0e22a254aa88/src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs#L970-L971

Reply:
> this is already asserted below in GetArgumentRefKind:

I know. But in order to find that, I had to debug the test. I also was curios what node is used for that extra argument. Needed to understand that in order to confirm that relaxation has the right logic and is appropriate. So, strengthening the assert will also benefit a future reader of the code.